app/src/main/java/cm/aptoide/pt/app/view/AppViewPresenter.java

```diff
@@ -501,11 +486,13 @@ private void sendSimilarAppsAdImpressionEvent(SimilarAppsViewModel similarAppsVi
         }, throwable -> crashReport.log(throwable));
   }
 
-  private void handleOnScroll() {
+  private void handleOnSimilarAppsVisible() {
     view.getLifecycleEvent()
         .filter(event -> event.equals(View.LifecycleEvent.CREATE))
-        .flatMap(lifecycleEvent -> view.scrollVisibleSimilarApps())
-        .takeUntil(__ -> view.isSimilarAppsVisible())
+        .flatMap(lifecycleEvent -> Observable.merge(view.scrollVisibleSimilarApps()
+            .takeUntil(__ -> view.isSimilarAppsVisible())
+            .map(__ -> true), view.similarAppsVisibilityFromInstallClick()))
```

Comment:
Couldn't we simply do `Observable.merge(view.scrollVisibleSimilarApps(), view.similarAppsVisibilityFromInstallClick())`, since we have first() afterwards? Why takeUntil & map?

Reply:
Yeah are right, will test and change it.